### PR TITLE
x/twap: track pool state changes after exit pool hooks

### PR DESCRIPTION
**Pull request by AlpinYukseloglu**
Closes: #2540 & #2546

## What is the purpose of the change

This PR makes the twap `AfterExitPool` listener track pool state changes. It also fixes a previously buggy listener test that conflated initial pool creation with the pool state changes being tested.

## Brief Changelog

- Add `trackChangedPool` to the twap `AfterExitPool` listener & add tests for it
- Fix and refactor `TestPoolStateChange`

## Testing and Verifying

This change added tests and can be verified as follows:

`AfterExitPool` listener tracker:
- Added tests in that ensure pool exits flag the pool as changed

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (no)
  - How is the feature or change documented? (not documented)

### CODE REVIEW

File: x/twap/listeners_test.go [hunk 1]
```diff
@@ -285,7 +300,13 @@ func (s *TestSuite) TestAfterSwap_JoinPool() {
...
 			}
 
-			// test that either of swapping in a pool or joining a pool
+			if tc.exitPool {
+				poolCreator := s.TestAccs[0]
+				lpBalance := s.App.BankKeeper.GetBalance(s.Ctx, s.TestAccs[0], "gamm/pool/1").Amount
+				s.App.GAMMKeeper.ExitPool(s.Ctx, poolCreator, poolId, lpBalance.Quo(sdk.NewInt(10)), sdk.Coins{})
+			}
+
+			// test that either of swapping in a pool, joining a pool, or exiting a pool
```
Comment (czarcas7ic) [inline x/twap/listeners_test.go:309]: Do you think we should make a `s.RunBasicExit` in the test suite that we can use for future tests?

Comment (ValarDragon) [reply x/twap/listeners_test.go:309]: That would be great to do

**Review by ValarDragon — APPROVED** [review]
Nice! Lets do Adam's suggestion of making a helper in apptesting, and then merge?

File: x/twap/listeners_test.go (hunk 1 above)
Comment (AlpinYukseloglu) [reply x/twap/listeners_test.go]: Done!

**Review by p0mvn — APPROVED** [review]

**Pull request merged**